File: 495. Teemo Attacking/solution.py
class Solution(object):
  def findPoisonedDuration(self, timeSeries, duration):
    """
    :type timeSeries: List[int]
    :type duration: int
    :rtype: int
    """
    totalTime = 0
    nextTime = 0
    for time in timeSeries:
      subtractedTime = 0
      if time < nextTime:
        subtractedTime = nextTime - time
      nextTime = time + duration
      totalTime += duration - subtractedTime
    return totalTime

File: 495. Teemo Attacking/solution.java
class Solution {
  public int findPoisonedDuration(int[] timeSeries, int duration) {
    int totalTime = 0;
    int nextTime = 0;
    for (int index = 0; index < timeSeries.length; index++){
      int time = timeSeries[index];
      int subtractedTime = 0;
      if (time < nextTime){
        subtractedTime = nextTime - time;
      }
      nextTime = time + duration;
      totalTime += duration - subtractedTime;
    }

    return totalTime;
  }
}